nlprule/src/rule/grammar.rs

```diff
@@ -75,30 +75,27 @@ impl PosReplacer {
                     .get_group_members(&x.lemma.as_ref().to_string());
                 let mut data = Vec::new();
                 for word in group_words {
-                    if let Some(i) = tokenizer
+                    if let Some(_i) = tokenizer
                         .tagger()
                         .get_tags(word)
                         .iter()
                         .position(|x| self.matcher.is_match(&x.pos))
                     {
-                        data.push((word.to_string(), i));
+                        data.push(word.to_string());
                     }
                 }
                 data
             })
             .rev()
             .flatten()
             .collect();
-        candidates.sort_by(|(_, a), (_, b)| a.cmp(b));
-        if candidates.is_empty() {
-            None
-        } else {
-            Some(candidates.remove(0).0)
-        }
+        candidates.sort_unstable();
+        candidates.dedup();
```

Comment:
This is probably the way to go: https://docs.rs/itertools/0.10.0/itertools/fn.kmerge.html